pallets/msa/src/lib.rs

```diff
@@ -393,8 +391,11 @@ pub mod pallet {
 				.map_err(|_| Error::<T>::AddKeySignatureVerificationFailed)?;
 
 			let msa_id = add_key_payload.msa_id;
+
 			Self::ensure_msa_owner(&who, msa_id)?;
 
+			Self::is_key_taken(msa_id, &key)?;
```

Comment:
https://github.com/LibertyDSNP/frequency/pull/308/files#diff-a15f544dbfef959186da2f9b4e7d8bd27aae8901824537de4f260e7a9b4f6c3bR506

I think this is checked already here.

Reply:
We also need to check if keyinfo msa_id == msa_id of who is trying to add that key, if not this a cross registration if yes we continue and next step i.e. try to add_key and returns duplicate key error

Reply:
Should the name of this function be refactored to something like `ensure_key_is_available` or `ensure_key_is_not_taken` since it returns an Ok value if the key is not taken?

Reply:
Or, like @saraswatpuneet 's comment says below we could move this logic check out of its own function and into `add_key`